packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -93,7 +100,32 @@ class _FlutterPlatform extends PlatformPlugin {
     this.explicitObservatoryPort,
     this.host,
     this.previewDart2,
-  }) : assert(shellPath != null);
+  }) : assert(shellPath != null) {
+    compilerController.stream.listen((CompilationRequest request) async {
+      bool isEmpty = compilationQueue.isEmpty;
+      compilationQueue.add(request);
+      // Only trigger processing if queue was empty - i.e. no other requests
+      // we being processed. This effectively enforces "one compilation request at a time".
+      if (isEmpty) {
+        while (!compilationQueue.isEmpty) {
+          CompilationRequest request = compilationQueue.first;
+          print("Compiling ${request.path}");
+          String outputPath = await compiler.recompile(request.path,
+            <String>[request.path]
+          );
+          print("Finished compilation of ${request.path} into $outputPath");
+          compiler.accept();
+          compiler.reset();
+          request.result.complete(outputPath);
+          // Only remove now when we finished processing the element
```

Comment:
in what circumstance would we try recompiling the same test again ? I am wondering why is is necessary to remove the request only after we are done with the compilation.

Reply:
It's not that we are going to recompile the same test again. We only remove the test from the queue after we are done compiling because that way we know that no other compilation request can start before we exit(no await statements from this point on).